crates/typstyle-core/src/partial.rs

```diff
@@ -0,0 +1,126 @@
+use std::ops::Range;
+
+use typst_syntax::{
+    ast::{Expr, Markup},
+    LinkedNode, Source, Span, SyntaxKind,
+};
+
+use crate::{pretty::Mode, utils, AttrStore, FormatError, PrettyPrinter, Typstyle, TypstyleConfig};
+
+impl Typstyle {
+    /// Format the node with minimal span that covering the given *byte* range.
+    /// If you have a utf16 range instead, use [`utf16_range_to_byte_range`] to construct the byte range.
+    pub fn format_byte_range(
+        &self,
+        source: &Source,
+        byte_range: Range<usize>,
```

Comment:
rename `byte_range` to `utf8_range` and then we can strip out comment "If you have a utf16 range instead, use [`utf16_range_to_byte_range`] to construct the byte range."